pkg/virt-controller/watch/migration.go

```diff
@@ -1346,6 +1390,62 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
 	return nil
 }
 
+func (c *MigrationController) addDeschedulerAnnotationToSourcePod(sourcePod *k8sv1.Pod) error {
+	if _, exists := sourcePod.GetAnnotations()[deschedulerEvictionInProgressAnnotation]; !exists {
+		patchSet := patch.New(
+			patch.WithAdd(fmt.Sprintf("/metadata/annotations/%s", patch.EscapeJSONPointer(deschedulerEvictionInProgressAnnotation)), ""),
+		)
+		patchBytes, err := patchSet.GeneratePayload()
+		if err != nil {
+			return err
+		}
+
+		_, err = c.clientset.CoreV1().Pods(sourcePod.Namespace).Patch(context.Background(), sourcePod.Name, types.JSONPatchType, patchBytes, v1.PatchOptions{})
+		if err != nil {
+			log.Log.Object(sourcePod).Errorf("failed to add %s pod annotation: %v", deschedulerEvictionInProgressAnnotation, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *MigrationController) removeDeschedulerAnnotationFromSourcePod(migration *virtv1.VirtualMachineInstanceMigration) error {
+	if migration.Status.MigrationState == nil || migration.Status.MigrationState.SourcePod == "" {
+		return nil
+	}
+
+	podKey := controller.NamespacedKey(migration.Namespace, migration.Status.MigrationState.SourcePod)
+	obj, exists, err := c.podIndexer.GetByKey(podKey)
+	if !exists {
+		log.Log.Reason(err).Errorf("source pod %s does not exist", migration.Status.MigrationState.SourcePod)
+		return nil
+	}
+	if err != nil {
+		log.Log.Reason(err).Errorf("Failed to fetch source pod %s for namespace from cache.", migration.Status.MigrationState.SourcePod)
+		return err
+	}
+
+	sourcePod := obj.(*k8sv1.Pod)
+	if _, exists := sourcePod.GetAnnotations()[deschedulerEvictionInProgressAnnotation]; exists {
+		patchSet := patch.New(
+			patch.WithTest(fmt.Sprintf("/metadata/annotations/%s", patch.EscapeJSONPointer(deschedulerEvictionInProgressAnnotation)), ""),
+			patch.WithRemove(fmt.Sprintf("/metadata/annotations/%s", patch.EscapeJSONPointer(deschedulerEvictionInProgressAnnotation))),
+		)
+		patchBytes, err := patchSet.GeneratePayload()
+		if err != nil {
+			return err
+		}
+
+		_, err = c.clientset.CoreV1().Pods(sourcePod.Namespace).Patch(context.Background(), sourcePod.Name, types.JSONPatchType, patchBytes, v1.PatchOptions{})
+		if err != nil {
+			log.Log.Object(sourcePod).Errorf("failed to remove %s pod annotation : %v", deschedulerEvictionInProgressAnnotation, err)
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
Actually maybe the consts and these 2 functions could be moved to a new `descheduler.go`?
I find `migration.go` to be already big and complicated enough...